src/tools/clippy/tests/ui/incompatible_msrv.stderr

```diff
@@ -36,17 +36,26 @@ LL |     my_panic!("foo");
    |
    = note: this error originates in the macro `my_panic` (in Nightly builds, run with -Z macro-backtrace for more info)
 
+error: sub-expression diverges
+  --> tests/ui/incompatible_msrv.rs:59:13
+   |
+LL |     assert!(core::panicking::panic("out of luck"));
+   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = note: `-D clippy::diverging-sub-expression` implied by `-D warnings`
+   = help: to override `-D warnings` add `#[allow(clippy::diverging_sub_expression)]`
```

Comment:
@rust-lang/clippy would this be considered a regression? The lint triggers here now because of the change to the expansion of `assert!`.

Reply:
It looks like a lint ordering issue (lint order is not defined). I would add `#[allow(clippy::diverging_sub_expression)]` (as this is not the lint this test wants to trigger) and check that the original diagnostic still stands.

Reply:
I'd rather allow unrelated lints.

I agree that cross-checking lints would be nice. But doing that in some tests won't give us any guarantees, but just increases the number of files that need to be updated for changes.

Reply:
From my experience on rustc, the number of errors being affected by changes isn't that big of a deal and has helped caught issues during development, but changed this test to avoid it, leaving it to the team to decide whether to add cross-lint tests at your leisure.